internal/pkg/agent/program/spec.go

```diff
@@ -40,11 +40,25 @@ type Spec struct {
 	When                  string               `yaml:"when"`
 	Constraints           string               `yaml:"constraints"`
 	RestartOnOutputChange bool                 `yaml:"restart_on_output_change,omitempty"`
-	ExprtedMetrics        []string             `yaml:"exported_metrics,omitempty"`
+	ExportedMetrics       []string             `yaml:"exported_metrics,omitempty"`
+	Process               *ProcessSettings     `yaml:"process,omitempty"`
+}
+
+// ProcessSettings process specific settings
+type ProcessSettings struct {
+	// Allows to override the agent stop timeout settings and specify a different stop timeout for Endpoint service
+	StopTimeoutSecs int `yaml:"stop_timeout"`
```

Comment:
Sure, can switch to time.Duration, just make sure not to use number without time units otherwise 180 will be converted to 180ns time.Duration. So the spec would have to have time with units, for example:
```
process:
    stop_timeout: 3m
```